otx/algorithms/detection/adapters/mmdet/utils/config_utils.py

```diff
@@ -177,20 +190,55 @@ def patch_adaptive_repeat_dataset(
             data_train.times = new_repeat
 
 
-@check_input_parameters_type({"train_dataset": DatasetParamTypeCheck, "val_dataset": DatasetParamTypeCheck})
+@check_input_parameters_type()
+def align_data_config_with_recipe(
+    data_config: ConfigDict,
+    config: Union[Config, ConfigDict]
+):
+    data_config = data_config.data
+    config = config.data
+    for subset in data_config.keys():
+        subset_config = data_config.get(subset, {})
+        for key in list(subset_config.keys()):
+            found_config = get_configs_by_keys(
+                config.get(subset),
+                key,
+                return_path=True
+            )
+            assert len(found_config) == 1
+            value = subset_config.pop(key)
+            path = list(found_config.keys())[0]
+            update_config(subset_config, {path: value})
```

Comment:
Could you shortly explain what this function is doing?

Reply:
`self._data_cfg` is always initialized in [this](https://github.com/cih9088/training_extensions/blob/feature/otx-new-mmcv/mmseg/otx/algorithms/detection/tasks/nncf.py#L360-L373) structure. If data configuration of recipe has a wrapper dataset, `self._data_cfg` structure must be changed accordingly to [update recipe configuration](https://github.com/cih9088/training_extensions/blob/feature/otx-new-mmcv/mmseg/otx/algorithms/detection/adapters/mmdet/utils/config_utils.py#L228-L237).

Reply:
This would be refactored in `mmcls` PR.